Mathlib/GroupTheory/Submonoid/Membership.lean

```diff
@@ -452,6 +452,30 @@ theorem powers_one : powers (1 : M) = ⊥ :=
   bot_unique <| powers_subset (one_mem _)
 #align submonoid.powers_one Submonoid.powers_one
 
+/-- The submonoid generated by an element is a group if that element has finite order. -/
+abbrev groupOfPowers {x : M} {n : ℕ} (hpos : 0 < n) (hx : x ^ n = 1) : Group (powers x) where
+  inv x := x ^ (n - 1)
+  mul_left_inv y := Subtype.ext <| by
+    obtain ⟨_, k, rfl⟩ := y
+    dsimp
```

Comment:
This is a non-terminal (d)simp and should be squeezed / captured with `suffices` or dealt with in some other way. See:

https://leanprover-community.github.io/extras/simp.html#non-terminal-simps
https://leanprover-community.github.io/extras/simp.html#dsimp

Reply:
All done, but the `squeeze_simp` tactic doesn't exist in Lean 4 so I had to do it manually!

Reply:
It's just `simp?` in Lean 4 ...

Reply:
Well I did use `simp?`